src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,54 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  static DEAD_VALUE = 0;
+  static HEALTH = 100;
+  static BITE_DAMAGE = 50;
+
+  health = Animal.HEALTH;
+
+  static removeDeadAnimals() {
+    this.alive = this.alive.filter((item) => item.health > this.DEAD_VALUE);
+  }
+
+  constructor(name) {
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  hidden = false;
+
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (animal instanceof Carnivore || animal.hidden) {
+      return;
```

Comment:
The condition `animal.hidden` will not work as expected if `animal` is not an instance of `Herbivore`, because `hidden` property is not defined in the `Animal` class. It's better to check if `animal` is an instance of `Herbivore` and then check if it's hidden.